test/integration/endpoints/endpoints_test.go

```diff
@@ -291,6 +294,20 @@ func TestEndpointWithTerminatingPod(t *testing.T) {
 
 	// poll until endpoint for deleted Pod is no longer in Endpoints.
 	if err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		// Ensure that the recently deleted Pod exists but with a deletion timestamp. If the Pod does not exist,
+		// we should fail the test since it is no longer validating against a terminating pod.
+		pod, err := client.CoreV1().Pods(ns.Name).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, fmt.Errorf("expected Pod %q to exist with deletion timestamp but was not found: %v", pod.Name, err)
+		}
+		if err != nil {
+			return false, nil
+		}
```

Comment:
no need for a change, but noting that this and the err check in the API call bellow are dropping the underlying error.
`t.Fatalf("error checking for no endpoints with terminating pods: %v", err)` would always print "timeout waiting for condition".

Reply:
you are right, but integration tests failures are easy to reproduce and is not a big issue like in the e2e tests,

Reply:
> t.Fatalf("error checking for no endpoints with terminating pods: %v", err) would always print "timeout waiting for condition".

> this is a problem in a lot of places though.

I've been mulling over this for a bit actually. If we return an error there, it cancels the poll which makes transient errors fail the test, leading to more flakes. Returning nil allows for retries but hides the error with just "timed out waiting for condition".

Reply:
i've seen it being stored as `lastError` in a few places.

```
if err != nil {
	lastError = err
	return false, nil
}

...

if err != nil {
	if lastError == nil {
		lastError = err
	}
	...
}
```

Reply:
yeah that's fair, but it would get so messy if we had to do that everytime we called wait.Poll :grimacing: